datafusion/substrait/src/logical_plan/producer/expr/literal.rs

```diff
@@ -94,6 +95,36 @@ pub(crate) fn to_substrait_literal(
             LiteralType::I64(*n as i64),
             UNSIGNED_INTEGER_TYPE_VARIATION_REF,
         ),
+        ScalarValue::Float16(Some(f)) => {
+            // Following rules from https://github.com/apache/arrow/blame/main/format/substrait/extension_types.yaml
```

Comment:
Perhaps a permalink would be better, something like

https://github.com/apache/arrow/blame/bab558061696ddc1841148d6210424b12923d48e/format/substrait/extension_types.yaml#L23